core/src/main/kotlin/org/lflang/generator/uc/UcReactorGenerator.kt

```diff
@@ -91,11 +91,12 @@ class UcReactorGenerator(private val reactor: Reactor, fileConfig: UcFileConfig,
         """.trimMargin()
     }
 
+    // FIXME: 3rd argument must be replaced with self->parent if not main reactor
     fun generateCtorDefinition() = with(PrependOperator) {
         """
             | void ${reactor.name}_ctor(${reactor.name} *self, Environment *env) {
             |  size_t trigger_idx = 0;
-            |  Reactor_ctor(&self->super, "${reactor.name}", env, ${if (numChildren > 0) "self->_children" else "NULL"}, $numChildren, ${if (reactor.reactions.size > 0) "self->_reactions" else "NULL"}, ${reactor.reactions.size}, ${if (numTriggers() > 0) "self->_triggers" else "NULL"}, ${numTriggers()});
+            |  Reactor_ctor(&self->super, "${reactor.name}", env, NULL, ${if (numChildren > 0) "self->_children" else "NULL"}, $numChildren, ${if (reactor.reactions.size > 0) "self->_reactions" else "NULL"}, ${reactor.reactions.size}, ${if (numTriggers() > 0) "self->_triggers" else "NULL"}, ${numTriggers()});
```

Comment:
@erlingrj I had to add `NULL` here to make it compile, but I assume it should be `self->parent` for all reactors that are not the `main` reactor. This might be needed on the base branch as well?

Reply:
Yes. Thanks. Unfortunately, I have made some changes to the reactor-uc API that are not reflected yet in the code-generator. I will try to bring everything up-to-date soon